src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -1310,6 +1410,64 @@ std::string ImageReplayer<I>::to_string(const State state) {
 }
 
 template <typename I>
+void ImageReplayer<I>::resync_image(Context *on_finish) {
+  dout(20) << dendl;
+
+  m_lock.Lock();
+  if (m_state == STATE_STOPPED) {
+
+    dout(20) << "replayer is already stopped, skip to delete image step of "
+             << "resync" << dendl;
+
+    int r = m_remote->ioctx_create2(m_remote_pool_id, m_remote_ioctx);
+    if (r < 0) {
+      derr << "error opening ioctx for remote pool " << m_remote_pool_id
+           << ": " << cpp_strerror(r) << dendl;
+      m_lock.Unlock();
+      return;
+    }
+    m_remote_journaler = new Journaler(m_threads->work_queue,
+                                     m_threads->timer,
+				     &m_threads->timer_lock, m_remote_ioctx,
+				     m_remote_image_id, m_local_mirror_uuid,
+                                     1);
+    r = m_remote_journaler->unregister_client();
+
+    delete m_remote_journaler;
+    m_remote_ioctx.close();
+    m_remote_journaler = nullptr;
+
+    if (r < 0 && r != -ENOENT) {
+      derr << "failed to unregister client in remote journal: "
+           << cpp_strerror(r) << dendl;
+      m_lock.Unlock();
+      return;
+    }
+
+    m_image_deleter->schedule_image_delete(m_local_pool_id,
+                                           m_local_image_id,
+                                           m_local_image_name,
+                                           m_global_image_id);
+
+    if (on_finish) {
+      m_image_deleter->wait_for_scheduled_deletion(m_local_image_name,
+                                                   on_finish, false);
+    }
+    m_lock.Unlock();
+  } else {
+    m_stopping_for_resync = true;
+    if (m_state == STATE_STARTING) {
```

Comment:
hmmm... so after merging Mykola PR, do you think that if I just call `stop`, it will get everything done?